library/src/main/java/com/novoda/downloadmanager/lib/DownloadNotifier.java

```diff
@@ -422,7 +422,7 @@ private void setSecondaryNotificationText(NotificationCompat.Builder builder, No
     }
 
     private void removeStaleTagsThatWereNotRenewed(SimpleArrayMap<String, Collection<DownloadBatch>> clustered) {
-        for (int i = 0, size = activeNotifs.size(); i < size; i++) {
+        for (int i = activeNotifs.size() - 1; i >= 0; i--) {
```

Comment:
why did it need reversing? Could you have just said `i <= size` to fix it also ?